simulariumio/writers/binary_writer.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+from typing import List, Tuple, Any, Dict, Union
+import struct
+import json
+
+import numpy as np
+
+from ..data_objects import (
+    AgentData,
+    TrajectoryData,
+)
+from ..constants import BINARY_SETTINGS, BINARY_BLOCK_TYPE, CURRENT_VERSION
+from .writer import Writer
+from .binary_chunk import BinaryChunk
+from .binary_values import BinaryValues
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class BinaryWriter(Writer):
+    @staticmethod
+    def _padding(n_bytes: int) -> int:
+        """
+        Return the number of bytes to pad, given the desired number.
+        """
+        remainder = n_bytes % BINARY_SETTINGS.BLOCK_OFFSET_BYTE_ALIGNMENT
+        if remainder > 0:
+            return BINARY_SETTINGS.BLOCK_OFFSET_BYTE_ALIGNMENT - remainder
+        else:
+            return 0
+
+    @staticmethod
+    def _frame_buffers_n_values(trajectory_data: TrajectoryData) -> List[int]:
+        """
+        Get the number of values in the bundle data buffer for each frame
+        """
+        total_steps = trajectory_data.agent_data.total_timesteps()
```

Comment:
If not changing this, then I think there needs to be some documentation somewhere that `frame` is synonymous with `timestep`, `step` and/or `time`.   When the code alternates between these terms interchangeably, I'd be worried that an engineer might be confused by lines like 
```
for frame_index in range(total_steps):
```
when it could easily be
```
for step_index in range(total_steps):
```

Reply:
I did change it `total_steps()` to `total_timesteps`. I could go through the codebase and remove `frame`, but I know it's used a bunch of places

Reply:
can be separate PR/issue but I think it deserves consistency.  Also see this thread for how another team discussed what the data of a timestep is: https://github.com/MDAnalysis/mdanalysis/issues/250#issuecomment-98279984